### PR TITLE
embedding/csharp: Add EventChannel and related classes

### PR DESCRIPTION
**Pull request by WonyoungChoi**
- #127 
- Add EventChannel and related classes
- Add tests for EventChannel

#### Usage
```c#
            var eventChannel = new EventChannel("test_event_channel");
            eventChannel.SetStreamHandler(new EventStreamHandler(
                (args, sink) =>
                {
                    Console.WriteLine("OnListen!!");
                    counter = 0;
                    timer = new Timer(100);
                    timer.Elapsed += (sender, e) =>
                    {
                        sink.Success("TEST_EVENT " + counter++);
                    };
                    timer.Start();
                },
                (args) =>
                {
                    Console.WriteLine("OnCancel!!");
                    timer.Stop();
                }));

```

### CODE REVIEW

File: embedding/csharp/Tizen.Flutter.Embedding/Channels/EventChannel.cs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                }
+                catch (Exception e)
+                {
+                    TizenLog.Error($"{_channel.Name}: Failed to close event stream. {e.Message}");
+                    return Task.FromResult(_channel.Codec.EncodeErrorEnvelope("error", e.Message, null));
+                }
+            }
+            else
+            {
+                return Task.FromResult(_channel.Codec.EncodeErrorEnvelope("error", "No active stream to cancle.", null));
```
Comment (swift-kim) [inline embedding/csharp/Tizen.Flutter.Embedding/Channels/EventChannel.cs:154]:
```suggestion
                return Task.FromResult(_channel.Codec.EncodeErrorEnvelope("error", "No active stream to cancel.", null));
```

File: embedding/csharp/Tizen.Flutter.Embedding/Channels/IEventSink.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        /// Consumes a successful event.
+        /// </summary>
+        /// <param name="event">The event, possibly null.</param>
+        void Success(object @event);
+
+        /// <summary>
+        /// Consumes an error event.
+        /// </summary>
+        /// <param name="code">An error code string.</param>
+        /// <param name="message">An human-readable error message string, possibly null.</param>
```
Comment (swift-kim) [inline embedding/csharp/Tizen.Flutter.Embedding/Channels/IEventSink.cs:22]:
```suggestion
        /// <param name="message">A human-readable error message string, possibly null.</param>
```

File: embedding/csharp/Tizen.Flutter.Embedding.Tests/Channels/EventChannelTests.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            }
+        }
+
+        public class TheSetStreamHandlerMethod
+        {
+            [Fact]
+            public void Registers_Stream_Handler()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                var handler = Substitute.For<IEventStreamHandler>();
```
Comment (swift-kim) [inline embedding/csharp/Tizen.Flutter.Embedding.Tests/Channels/EventChannelTests.cs:47]: How about using `new EventStreamHandler()` here?

File: embedding/csharp/Tizen.Flutter.Embedding/Channels/EventChannel.cs (hunk 1 above)
Comment (WonyoungChoi) [reply embedding/csharp/Tizen.Flutter.Embedding/Channels/EventChannel.cs:154]: good catch! thanks!

File: embedding/csharp/Tizen.Flutter.Embedding.Tests/Channels/EventChannelTests.cs (hunk 3 above)
Comment (WonyoungChoi) [reply embedding/csharp/Tizen.Flutter.Embedding.Tests/Channels/EventChannelTests.cs:47]: This test to ensure that the call is delivered properly to the `BinaryMessenger`. IMO, it is not necessary to create an `EventStreamHandler` instance directly. `Test double` is enough in this case.

**Review by swift-kim — APPROVED** [review]
Good work!

**Pull request merged**